api/app/signals/apps/api/serializers/attachment.py

```diff
@@ -24,12 +25,22 @@
 
 class SignalAttachmentSerializerMixin:
     def create(self, validated_data):
-        attachment = Signal.actions.add_attachment(validated_data['file'], self.context['view'].get_signal())
+        user = self.context['request'].user
+        signal = self.context['view'].get_signal()
 
-        if self.context['request'].user:
-            attachment.created_by = self.context['request'].user.email
+        # save our attachment
+        attachment = Signal.actions.add_attachment(validated_data['file'], signal)
+        if user:
+            attachment.created_by = user.email
             attachment.save()
 
+        # add a note that an attachment (currently only images allowed) was uploaded
+        filename = os.path.basename(attachment.file.name)
+        msg = f'Foto toegevoegd door melder: {filename}'
+        if user:
+            msg = f'Foto toegevoegd door {user.email}: {filename}'
```

Comment:
Should we add a fallback, if for some reason the `user.email` is not set this will not work.
Could be something like `{user.email or user.username}`

Reply:
Ticket will be made to fix this globally in the API

Reply:
There are more places where Signalen expects the email address to be available (we inherited this from the old authentication system in Amsterdam). A lot of our logging and our `created_by` fields are set to `user.email`.